src/main/cpp/src/HyperLogLogPlusPlusHostUDFJni.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024-2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cudf_jni_apis.hpp"
+#include "hyper_log_log_plus_plus.hpp"
+#include "hyper_log_log_plus_plus_host_udf.hpp"
+
+#include <cudf/aggregation/host_udf.hpp>
+
+extern "C" {
+
+JNIEXPORT jlong JNICALL
+Java_com_nvidia_spark_rapids_jni_HyperLogLogPlusPlusHostUDF_createHLLPPHostUDF(JNIEnv* env,
+                                                                               jclass,
+                                                                               jint agg_type,
+                                                                               int precision)
+{
+  try {
+    cudf::jni::auto_set_device(env);
```

Comment:
I don't think we will need to set device when creating the UDF instance, as we don't start the computation yet.
```suggestion
```

Reply:
Not sure about this. 
If it's removed, then there is no set device any more in this file, in general, there should be at least one set device.

Reply:
We only set device when we are about to run CUDA code. Here we don't run any, just create an instance of the UDF functor thus don't need it.